### ISSUE TITLE
Transactions are reset after restart

### ISSUE THREAD

**Issue by NejcZdovc** [brave/brave-browser#6745]
## Steps to Reproduce

   1.  claim promotion
   2. tip 3 times 5 BAT to verified publisher
   3. wait for transaction to be completely  processed
   4. open ledger_state 
   5. make sure that transaction data is in there
   6. restart browser
   7. open ledger_sate

## Actual result:

Transaction object is empty

```
"transactions": [
    {
      "viewingId": "",
      "surveyorId": "",
      "rates": {},
      "contribution_probi": "",
      "submissionStamp": "",
      "anonizeViewingId": "",
      "registrarVK": "",
      "masterUserToken": "",
      "surveyorIds": [],
      "votes": 0,
      "ballots": []
    }
  ],
```

## Expected result:
Transaction object should persist after restart

## Version/Channel Information:

- Can you reproduce this issue with the current release?
- Can you reproduce this issue with the beta channel?
- Can you reproduce this issue with the dev channel?
- Can you reproduce this issue with the nightly channel?

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields?
- Does the issue resolve itself when disabling Brave Rewards?
- Is the issue reproducible on the latest version of Chrome?

**Comment by jhoneycutt** [comment]
https://github.com/brave/brave-core/commit/0177ac5c42c6fddb315b918051dafc3ada91e97e removed some unused values from being read from/written to `ledger_state`, but it did not remove the code that _checked for_ them `TRANSACTION_ST::loadFromJson`.

**Comment by NejcZdovc** [comment]
as @jhoneycutt said it regressed with https://github.com/brave/brave-browser/issues/5917 which landed in 0.71

**Comment by LaurenWags** [comment]
Verified passed with

Brave | 0.71.114 Chromium: 78.0.3904.97 (Official Build) (64-bit)
-- | --
Revision | 021b9028c246d820be17a10e5b393ee90f41375e-refs/branch-heads/3904@{#859}
OS | macOS Version 10.13.6 (Build 17G5019)

* Verified test plan from https://github.com/brave/brave-core/pull/3917
* Verified STR from description
* Also verified after each time #6607 was tested in https://github.com/brave/brave-browser/issues/6607#issuecomment-552580909
